### PR TITLE
Doc about create a custom choices list

### PR DESCRIPTION
**Pull request by anaelChardan**

### CODE REVIEW

File: src/Acme/Bundle/DummyConnectorBundle/Resources/config/form_extensions.yml [hunk 1]
```diff
@@ -327,3 +327,16 @@ extensions:
             identifier:
                 path: code
                 name: code
+
+# Resources/config/form_extensions/acme_job_instance_csv_base_import.yml
+
+acme-job-instance-csv-base-import-show-properties-my-custom-list:
+        module: %THE_NAME_OF_YOUR_FECTHER%
```
Comment (mickaelandrieu) [inline src/Acme/Bundle/DummyConnectorBundle/Resources/config/form_extensions.yml]:
```diff
- %foo%
+ "%foo%"
```

File: src/Acme/Bundle/DummyConnectorBundle/Resources/config/form_extensions.yml [hunk 2]
```diff
@@ -327,3 +327,16 @@ extensions:
...
+        parent: acme-job-instance-csv-base-import-show-properties
+        position: 155
+        targetZone: global-settings
+        config:
+            fieldCode: configuration.myCustomList
+            readOnly: false
+            label: %your_label_translation_key%
```
Comment (mickaelandrieu) [inline src/Acme/Bundle/DummyConnectorBundle/Resources/config/form_extensions.yml]:
```diff
- %foo%
+ "%foo%"
```

File: src/Acme/Bundle/DummyConnectorBundle/Resources/config/form_extensions.yml [hunk 3]
```diff
@@ -327,3 +327,16 @@ extensions:
...
+            tooltip: %your_tooltip_translation_key%
```
Comment (mickaelandrieu) [inline src/Acme/Bundle/DummyConnectorBundle/Resources/config/form_extensions.yml]: same here

File: src/Acme/Bundle/DummyConnectorBundle/Resources/config/requirejs.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Resources/config/requirejs.yml
+
+config:
+paths:
+    %THE_NAME_OF_YOUR_FECTHER% : %the_path_of_your_fetcher% #For example -> mycustomfetcher: mycustombundle/js/my-custom-choices-list
```
Comment (mickaelandrieu) [inline src/Acme/Bundle/DummyConnectorBundle/Resources/config/requirejs.yml]: same here

File: src/Acme/Bundle/DummyConnectorBundle/Resources/config/requirejs.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+config:
+    pim/fetcher-registry:
+        fetchers:
+            %THE_NAME_OF_YOUR_FECTHER% :
```
Comment (mickaelandrieu) [inline src/Acme/Bundle/DummyConnectorBundle/Resources/config/requirejs.yml]: same here

File: src/Acme/Bundle/DummyConnectorBundle/Resources/config/requirejs.yml [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+                module: pim/base-fetcher
+                options:
+                    urls:
+                        list: %the_route_name_of_your_controller_action%
```
Comment (mickaelandrieu) [inline src/Acme/Bundle/DummyConnectorBundle/Resources/config/requirejs.yml]: same here

File: cookbook/import_export/create-connector.rst [hunk 7]
```diff
@@ -151,11 +151,12 @@ We need to provide a form name to the frontend to be able to render it properly.
     :linenos:
     :lines: 1-6
 
-If you want to go further, you can create your own form and re-use common parts provided by default. Here is an example of a full form configuration:
+If you want to go further, you can create your own form and re-use common parts provided by default. Here is an example of a full form configuration (config/form_extensions/acme_job_instance_csv_base_import.yml):
```
Comment (mickaelandrieu) [inline cookbook/import_export/create-connector.rst]: Can you create a link, or put the content?

File: cookbook/import_export/create-connector.rst [hunk 8]
```diff
@@ -164,6 +165,87 @@ Register it in the form provider and you are good to go!
...
+
+        public function listAction() {
+            $choices['my_option_1'] = 'my_option_1';
+            $choices['my_option_2'] = 'my_option_2';
+
+            return new JsonResponse($choices);
+        }
+    }
+
+Then configure a fetcher
```
Comment (mickaelandrieu) [inline cookbook/import_export/create-connector.rst]: a fetcher ?

Also:

```diff
- a fetcher
+ a fetcher:
```

File: cookbook/import_export/create-connector.rst (hunk 7 above)
Comment (anaelChardan) [inline cookbook/import_export/create-connector.rst]: It's just to tell the path to the user, I don't know how to do that :)

File: cookbook/import_export/create-connector.rst (hunk 8 above)
Comment (anaelChardan) [inline cookbook/import_export/create-connector.rst]: Ok :)

File: cookbook/import_export/create-connector.rst [hunk 9]
```diff
@@ -164,6 +165,87 @@ Register it in the form provider and you are good to go!
...
+
+.. literalinclude:: ../../src/Acme/Bundle/DummyConnectorBundle/Resources/config/requirejs.yml
```
Comment (anaelChardan) [inline cookbook/import_export/create-connector.rst]: Is this that create an error ? ok thanks :)

File: src/Acme/Bundle/DummyConnectorBundle/Resources/config/form_extensions.yml (hunk 1 above)
Comment (anaelChardan) [inline src/Acme/Bundle/DummyConnectorBundle/Resources/config/form_extensions.yml]: Yeah :)

File: cookbook/import_export/create-connector.rst (hunk 9 above)
Comment (mickaelandrieu) [inline cookbook/import_export/create-connector.rst]: Yup, the Python validator for YAML files is "stricter" than Symfony 2 one: note that in Symfony 3 this syntax is also invalid:

```yaml
foo: %foo% # valid in Symfony2, invalid in Symfony3 and in YAML reference
foo: "%foo%" # valid everywhere
```

**Comment by mickaelandrieu** [comment]
Thank you @anaelChardan 👍

**Comment by anaelChardan** [comment]
Thanks to you @mickaelandrieu  :)

**Pull request merged**